### ISSUE TITLE
[Question] Is Playwright Codegen the best auto-recorder on the market?

### ISSUE THREAD

**Issue by ufst-ask** [microsoft/playwright#20030]
Hi again.
Multiple tools offer some version of "auto-record". Testim, Leapwork, Cypress Studio, Macro Recorder, Playwright, Functionize etc.

I believe there is a great deal of value to this, just yesterday I tested codegen against the various types of componnts int the Float UI library via https://www.floatui.com/components/ .
It could auto-record over 90% which I think is very good.

Do you have, or know of, any comparison of "auto-record reliability" across the different tools out there?

If not, I might make one myself :)

**Comment by aslushnikov** [comment]
> Do you have, or know of, any comparison of "auto-record reliability" across the different tools out there?

@ufst-ask No, we are not aware of any. Feel free to do one yourself!

**Comment by ufst-ask** [comment]
@aslushnikov I will :)
It'll probably be a few weeks before I get the time to do it, but then I'll post it here and let you know.

**Comment by mxschmitt** [comment]
> @aslushnikov I will :)
> 
> It'll probably be a few weeks before I get the time to do it, but then I'll post it here and let you know.

Feel also free to contribute to our recorder, since we learned creating an own recorder is not the easiest thing due a lot of edge cases where you need to be aware of.

(personal opinion) Generally speaking I can say that the official Playwright codegen is the best for Playwright since it's making all it calls via Playwright to ensure the actual generated script works as expected. Also it uses the Playwright locator best practices aka. GetBy*

**Comment by ufst-ask** [comment]
@mxschmitt 
Yes. At the moment I don't know how, but I'll look into when it becomes relevant.
For his specific case I don't think it's much of an edge-case, more that codegen tried to compare on a long string which were a combination of header and paragraph.
Which I, as a quick fix, changed to look at less text through a HasText:

`//FAIL: //page.frameLocator("#iframe >> nth=0").getByRole(AriaRole.LINK, new FrameLocator.GetByRoleOptions().setName("What is SaaS? Software as a Service Explained Sidi dev Sidi dev Jan 4 2022 What is SaaS? Software as a Service Explained Going into this journey, I had a standard therapy regimen, based on looking at the research literature. After I saw the movie, I started to ask other people what they did for their anxiety, and some")).click();
            //Fixed version:
            page.frameLocator("#iframe >> nth=0").getByRole(AriaRole.LINK).filter(new Locator.FilterOptions().setHasText("What is SaaS")).click();`
